plotting_functions.py

```diff
@@ -0,0 +1,1653 @@
+    im : str, optional
+        Intensity measure to be used in the plots. Default is "PGA".
+    vs30 : int, optional
+        Vs30 value to be used in the plots. Default is 400.
+    im_xlims : tuple, optional
+        X-axis limits for the intensity measure. Default is (9e-5, 5).
+    poe_min_plot : float, optional
+        Minimum probability of exceedance to be plotted. Default is 1e-5.
+    ybuffer_absmax_over_val : float, optional
+        Defines how much to extend the y-axis limits above the maximum value in the plot (for a buffer).
+        The amount of extension is defined as the maximum plotted value divided by this value. Default is 10.0.
+    plot_dpi : int, optional
+        DPI for the generated plots. Default is 500.
+
+    Returns
+    -------
+    None
+    """
+
+    locations_nloc_dict = toml.load("resources/location_code_to_nloc_str.toml")
+    model_name_to_plot_format = toml.load("resources/model_name_lookup_for_plot.toml")
+    nshm_im_levels = np.loadtxt("resources/nshm_im_levels.txt")
+
+    needed_im_level_indices = np.where(
+        (nshm_im_levels >= im_xlims[0]) & (nshm_im_levels <= im_xlims[1])
+    )[0]
+
+    if isinstance(logic_tree_index_dir, str):
+        logic_tree_index_dir = Path(logic_tree_index_dir)
+    if isinstance(plot_output_directory, str):
+        plot_output_directory = Path(plot_output_directory)
+
+    if not plot_output_directory.exists():
+        plot_output_directory.mkdir(parents=True)
+
+    individual_realization_df = (
+        ds.dataset(
+            source=logic_tree_index_dir / "individual_realizations", format="parquet"
+        )
+        .to_table()
+        .to_pandas()
+    )
+
+    output_notes = toml.load(logic_tree_index_dir / "notes.toml")
+
+    if (
+        output_notes["source_logic_tree_note"] == "full > "
+        and output_notes["ground_motion_logic_tree_note"] == "full > "
+    ):
+        model_name_short = "full"
+        model_name_long = model_name_short
+
+    if (
+        output_notes["source_logic_tree_note"] == "full > 1 (nth) h.w.b. > "
+        and output_notes["source_logic_tree_note"] == "full > "
+    ):
+        model_name_short = "all_srm"
+        model_name_long = model_name_short
+
+    if (
+        output_notes["source_logic_tree_note"] == "full > "
+        and output_notes["source_logic_tree_note"] == "full > 1 (nth) h.w.b. > "
+    ):
+        model_name_short = "all_gmcm"
+        model_name_long = model_name_short
+
+    ### The GMCM logic tree was reduced to the single highest weighted branch so we can use the SRM model components
+    if "1 (nth) h.w.b." in output_notes["ground_motion_logic_tree_note"]:
+        model_name_short = (
+            output_notes["source_logic_tree_note"]
+            .split(">")[1]
+            .split("[")[1]
+            .strip("[ ] ")
+            + "_"
+            + output_notes["source_logic_tree_note"].split(">")[-2].strip()
```

Comment:
This should probably be a regex. It's very hard to figure out what you are matching here.